packages/core/src/StateNode.ts

```diff
@@ -182,13 +182,9 @@ export class StateNode<
     this.history =
       this.config.history === true ? 'shallow' : this.config.history || false;
 
-    this.entry = toArray(this.config.entry).map((action) =>
-      toActionObject(action, this.machine.options.actions)
-    );
+    this.entry = toActionObjects(this.config.entry);
+    this.exit = toActionObjects(this.config.exit);
```

Comment:
this is the core of the fix - we can't resolve those action objects here because their references in the `implementations` might change later on